adserver/views.py

```diff
@@ -1536,15 +1538,72 @@ def get_context_data(self, **kwargs):  # pylint: disable=too-many-locals
         return context
 
 
-class UpliftReportView(AllPublisherReportView):
+class UpliftReportView(BaseReportView):
 
     """An uplift report for all publishers."""
 
+    export_view = "publisher_uplift_report_export"
+    fieldnames = ["index", "views", "clicks", "ctr", "ecpm", "revenue", "our_revenue"]
     model = UpliftImpression
     force_revshare = 70.0
     report = PublisherUpliftReport
     template_name = "adserver/reports/all-publishers-uplift.html"
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        sort = self.request.GET.get("sort", "")
+        force_revshare = self.request.GET.get("force_revshare", self.force_revshare)
+
+        # Get all publishers where an ad has a view or click in the specified date range
+        impressions = self.get_queryset(
+            start_date=context["start_date"],
+            end_date=context["end_date"],
+        )
+        publishers = Publisher.objects.filter(id__in=impressions.values("publisher"))
+
+        revenue_share_percentage = self.request.GET.get("revenue_share_percentage", "")
+        if revenue_share_percentage:
+            try:
+                publishers = publishers.filter(
+                    revenue_share_percentage=float(revenue_share_percentage)
+                )
+            except ValueError:
+                pass
+
+        queryset = self.get_queryset(
```

Comment:
Calling `get_queryset` 2x in this function is a bit weird, I'm guessing just because we're filtering for publishers then doing the full data set?

Reply:
Ya. We could probably get rid of the 2nd call.

Reply:
I removed it